test/controllers/users_controller_test.js

```diff
@@ -241,6 +241,101 @@ describe('Endpoints exists for users', () => {
     });
   });
 
+  describe('Change user info', () => {
+    let userToken = null;
+    before((done) => {
+      chai.request(usersRoutes)
+        .post('/users')
+        .set('content-type', 'application/json')
+        .send({
+          email: 'userToBeChanged@email.com',
+          name: 'userToBeChanged',
+          password: 'passwordToBeChanged',
+        })
+        .end((err, res) => {
+          should.not.exist(err);
+          res.should.have.status(200);
+          chai.request(usersRoutes)
+            .post('/users/login')
+            .set('content-type', 'application/json')
+            .send({
+              email: 'userToBeChanged@email.com',
+              password: 'passwordToBeChanged',
+            })
+            .end((loginErr, loginRes) => {
+              should.not.exist(loginErr);
+              loginRes.should.have.status(200);
+              should.exist(loginRes.body.token);
+              userToken = loginRes.body.token;
+              done();
+            });
+        });
+    });
+
+    it('changes the user name successfully', (done) => {
```

Comment:
What about cases where they are unsuccessful?

Reply:
there is currently no way of testing the unsuccessful change - since username changes won't fail as a request unless there's a database error. Once we start unit testing - with mocking libraries, we will be able to test the unsuccessful case